src/csv.pp.ml

```diff
@@ -118,10 +120,26 @@ type in_channel = {
   (* Whitespace related stripping functions: *)
   is_space : char -> bool;
   lstrip_buffer : Buffer.t -> unit;
-  rstrip_substring : Bytes.t -> int -> int -> string;
+  rstrip_substring : IF_EIO(Cstruct.t, Bytes.t) -> int -> int -> string;
   rstrip_contents : Buffer.t -> string;
 }
 
+IF_EIO(
+let rstrip_substring buf off len =
+  let n = ref(off + len - 1) in
+  while !n >= off && is_space_or_tab(Cstruct.get buf !n) do decr n done;
+  Cstruct.to_string buf ~off ~len:(!n - off + 1)
+
+let in_buf_substring buf off len = Cstruct.to_string buf ~off ~len
+let unsafe_get_in_buf ({ buffer; off; _ } : Cstruct.t) i = Bigstringaf.unsafe_get buffer (off + i)
+let add_current_field_from_in_buf ic i =
+  Buffer.add_string ic.current_field (Cstruct.to_string ic.in_buf ~off:ic.in0 ~len:(i - ic.in0))
```

Comment:
Having read the source of Eio and Bigstringaf, this seems to be the most efficient way to copy from a `Cstruct.t` to a `Buffer.t`